app/makers_bnb.rb

```diff
@@ -0,0 +1,112 @@
+ENV["RACK_ENV"] ||= "development"
+
+require 'sinatra/base'
+require 'sinatra/flash'
+require_relative 'data_mapper_setup'
+
+class MakersBnB < Sinatra::Base
+  enable :sessions
+  register Sinatra::Flash
+  use Rack::MethodOverride
+  set :session_secret, 'super secret'
+
+  helpers do
+    def current_user
+      @current_user ||= User.get(session[:user_id])
+    end
+  end
+
+  post '/spaces/filter' do
+    session[:filter_from] = params[:filter_from]
+    session[:filter_to] = params[:filter_to]
+    redirect '/spaces_filtered'
+  end
+
+  get '/spaces_filtered' do
+    @spaces_old = Space.all
+    @spaces = []
+    @spaces_old.each do |space|
+      if Date.strptime(session[:filter_from], '%Y-%m-%d').to_time.to_i >= space.available_from.to_time.to_i && Date.strptime(session[:filter_to], '%Y-%m-%d').to_time.to_i <= space.available_to.to_time.to_i
+        @spaces << space
+      end
```

Comment:
This is messy. Your first thoughts should be 'how can I make this more readable?', and 'how can I push this further down the stack?'. Lack of refactoring here.